src/lib.rs

```diff
@@ -400,22 +410,52 @@ pub fn encode_to_fmt<T: AsRef<[u5]>>(
     }
 }
 
+/// Used for encode/decode operations for the two variants of Bech32
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Debug)]
+pub enum Variant {
+    /// The original Bech32 described in [BIP-0173](https://github.com/bitcoin/bips/blob/master/bip-0173.mediawiki)
+    Bech32,
```

Comment:
can we pls have the actual constants assigned as the values here? When we will not need `BECH32_CONST`, `BECH32M_CONST`,  and can simply use `Variant::Bech32 as u32` instead of `constant()` function call (and avoid one call)

Reply:
I'm fine with @clarkmoody's version, it's more "standard" imo, do you feel strongly about this?

Reply:
I'm not exactly sure what you mean @dr-orlovsky. Could you post a code example?

Performance-wise, the compiler will inline these simple methods for sure.